src/systems/acoustic_comms/AcousticComms.cc

```diff
@@ -61,8 +61,76 @@ class AcousticComms::Implementation
            std::tuple<std::chrono::duration<double>,
             std::chrono::duration<double>>>
           lastMsgReceivedInfo;
+
+  public: bool propagationModel(double _distToSource,
+                                int _numBytes);
+
+  /// \brief Source power in Watts.
+  /// \ref https://www.sciencedirect.com/topics/
+  /// computer-science/underwater-acoustic-signal
+  public: double sourcePower = 2000;
+
+  /// \brief Ratio of the noise intensity at the
+  /// receiver to the same reference intensity used for source level.
+  public: double noiseLevel = 1;
+
+  /// \brief Ratio of the total noise power at the array to the
+  /// noise received by the array along its main response axis.
+  /// \ref https://ieeexplore.ieee.org/document/5664178
+  public: double directivityIndex = 4;
+
+  /// \brief Information rate that can be transmitted over a given
+  /// bandwidth in a specific communication system, in (bits/sec)/Hz.
+  /// \ref: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5514747/
+  public: double spectralEfficiency = 7.0;
+
+  /// \brief Flag to store if the propagation model should be used.
+  public: bool usePropagationModel = false;
 };
 
+//////////////////////////////////////////////////
+bool AcousticComms::Implementation::propagationModel(
+  double _distToSource,
+  int _numBytes
+)
+{
+  // From https://www.mathworks.com/help/phased/ug/sonar-equation.html
+  // SNR = SL - TL - (NL - DI)
+  // SNR : Signal to noise ratio.
+  // SL : Source level. Ratio of the transmitted intensity from
+  //      the source to a reference intensity (1 m from source),
+  //      converted to dB.
+  // TL : Transmission loss (dB)
+  // NL : Noise level.
+  // DI : Receiver directivity index.
+
+  double sl = 171.5 + 10 * std::log10(this->sourcePower);
```

Comment:
Might be worth mentioning where 171.5 comes from. IIUC, this is the intensity measured at 1 yard (why use yards? :skull_and_crossbones:) from the source of the sound.

Reply:
Oh right just saw that. Changed it to 1m, updated docstring : https://github.com/gazebosim/gz-sim/pull/1793/commits/924e18e878bba1c14c1854b1d5612489951a2791